ulib/experimental/Steel.DisposableInvariant.fst

```diff
@@ -17,45 +17,38 @@ let conditional_inv (r:ghost_ref bool) (p:slprop) =
 let ex_conditional_inv (r:ghost_ref bool) (p:slprop) =
     h_exists (conditional_inv r p)
 
-let inv (p:slprop) = (r:ghost_ref bool & Steel.Memory.inv (ex_conditional_inv r p))
+let inv (p:slprop) = erased (r:ghost_ref bool & Steel.Memory.inv (ex_conditional_inv r p))
 
 let name (#p:_) (i:inv p) = dsnd i
 let gref (#p:_) (i:inv p) = dfst i
 
 [@@__reduce__]
 let active (#p:_) ([@@@smt_fallback]f:perm) (i:inv p) =
-    ghost_pts_to (gref i) (half_perm f) (hide true)
+  ghost_pts_to (gref i) (half_perm f) (hide true)
 
-let new_inv (#u:_) (p:slprop)
-  : SteelGhostT (inv p) u p (active full_perm)
-  = let r = ghost_alloc (Ghost.hide true) in
-    ghost_share r;
-    A.intro_exists true (conditional_inv r p);
-    let i = A.new_invariant u (ex_conditional_inv r p) in
-    (| r, i |)
+let new_inv #u p =
+  let r = ghost_alloc (Ghost.hide true) in
+  ghost_share r;
+  A.intro_exists true (conditional_inv r p);
+  let i = A.new_invariant u (ex_conditional_inv r p) in
+  rewrite_context ();
```

Comment:
Yes, to reduce `gref (hide (| r, x |))` to `r`.

Reply:
Ok. Could you add a TODO commenting this rewrite so that I remember where to find this once we look deeper into either reducing or rewriting reveal_hide/hide_reveal?